main.go

```diff
@@ -64,28 +61,22 @@ func main() {
 	go serveMetrics(cfg.MetricsAddress)
 	go handleSigterm(stopChan)
 
-	client, err := newClient(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	serviceSource, err := source.NewServiceSource(client, cfg.Namespace, cfg.FqdnTemplate, cfg.Compatibility)
-	if err != nil {
-		log.Fatal(err)
-	}
-	source.Register("service", serviceSource)
-
-	ingressSource, err := source.NewIngressSource(client, cfg.Namespace, cfg.FqdnTemplate)
-	if err != nil {
-		log.Fatal(err)
+	// Create a source.Config from the flags passed by the user.
+	sourceCfg := source.Config{
+		KubeMaster:    cfg.Master,
+		KubeConfig:    cfg.KubeConfig,
+		Namespace:     cfg.Namespace,
+		FQDNTemplate:  cfg.FQDNTemplate,
+		Compatibility: cfg.Compatibility,
 	}
-	source.Register("ingress", ingressSource)
 
-	sources, err := source.LookupMultiple(cfg.Sources)
+	// Lookup all the selected sources and pass them the desired configuration.
+	sources, err := source.LookupMultiple(cfg.Sources, &sourceCfg)
```

Comment:
Honestly, I wouldn't prefer splitting the config as it's nice to have a single config struct. Having two copies of the kubeClient is undesireable though. Also, copying the config would actually result in many copies, because we wrap Sources a couple of times by now (DedupSource, MultipleSource, ...). 

Bottom line: I personally prefer the pointer + read-only contract approach. It's not that we have arbitrary users adding sources and messing with our params. It's all in the source package and a reviewer should make sure the passed in config is not modified.

But I'm fine with either version. I leave the decison to you @ismith @ideahitme @Raffo @mikkeloscar.